docroot/themes/humsci/su_humsci_theme/js/hover_menu.js

```diff
@@ -29,25 +29,67 @@
         }
       }
 
+      // Check main menu items to ensure dropdown submenus don't get orphaned or cut off.
+      function menuEdgeCheck() {
+        var viewportWidth = document.body.clientWidth;
+        jQuery('ul.decanter-nav-primary > li').each(function() {
+          var itemFromLeft = jQuery(this).offset().left;
+          var itemFromRight = viewportWidth - itemFromLeft;
+          var subMenuWidth = jQuery('> ul', this).outerWidth();
+          if (subMenuWidth > itemFromRight) {
+            jQuery(this).addClass( 'edge' );
+          }
+          else {
+            jQuery(this).removeClass( 'edge' );
+          }
+        });
+      }
+
       setMenu();
-      $(window).resize(setMenu);
+      menuEdgeCheck();
+
+      var onResizeActivity = debounce(function() {
+        setMenu();
+        menuEdgeCheck();
+      }, 125);
 
       // Open/close the menu from hamburger button.
       $header.find('button.fa-bars').once().click(function() {
-        $(this).siblings('ul').toggleClass('expanded').find('.fa-minus').each(function() {
-          $(this).toggleClass('fa-plus').toggleClass('fa-minus');
-          $(this).siblings('ul').toggleClass('expanded');
-        });
+        menuExpander(this);
       });
 
       // Open/close submenus from the plus button.
       $header.find('button.fa-plus').once().click(function() {
-        $(this).siblings('ul').toggleClass('expanded').find('.fa-minus').each(function() {
-          $(this).toggleClass('fa-plus').toggleClass('fa-minus');
-          $(this).siblings('ul').toggleClass('expanded');
-        });
+        menuExpander(this);
         $(this).toggleClass('fa-plus').toggleClass('fa-minus');
       });
+
+      function menuExpander(theMenu) {
+        $(theMenu).siblings('ul').toggleClass('expanded').find('.fa-minus').each(function() {
+          $(theMenu).toggleClass('fa-plus').toggleClass('fa-minus');
+          $(theMenu).siblings('ul').toggleClass('expanded');
+        });
+      }
+
+      // debounce so that we don't run our resize functions constantly
+      function debounce(func, wait, immediate) {
```

Comment:
Debounce is included in D8 core. Please use it instead. see `drupal.debounce` in source at https://api.drupal.org/api/drupal/core%21core.libraries.yml/8.5.x

Reply:
@josephgknox @pookmish if you haven't please also review that list of libraries as there might be some interesting items you were unaware are in core.

Reply:
displace.js may be helpful for what you are trying to achieve here. https://www.drupal.org/node/1956804

Reply:
less code for us to maintain and more reliant on existing libraries

Reply:
Done.